operate/client/src/modules/stores/processInstanceListeners.ts

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import {makeAutoObservable} from 'mobx';
+import {
+  fetchProcessInstanceListeners,
+  ListenerPayload,
+  ListenersDto,
+} from 'modules/api/processInstances/fetchProcessInstanceListeners';
+
+type FetchType = 'initial' | 'prev' | 'next';
+
+type State = {
+  listenersCount: number;
+  listeners: ListenersDto['listeners'];
+  page: number;
+  latestFetch: {
+    fetchType: FetchType | null;
+    itemsCount: number;
+  };
+  status:
+    | 'initial'
+    | 'first-fetch'
+    | 'fetching'
+    | 'fetching-next'
+    | 'fetching-prev'
+    | 'fetched'
+    | 'error';
+  currentProcessInstanceId: ProcessInstanceEntity['id'];
+  currentFlowNodeId: string;
+};
+
+const DEFAULT_STATE: State = {
+  listenersCount: 0,
+  listeners: [],
+  page: 1,
+  latestFetch: {fetchType: null, itemsCount: 0},
+  status: 'initial',
+  currentProcessInstanceId: '',
+  currentFlowNodeId: '',
+};
+
+const MAX_LISTENERS_STORED = 200;
+const MAX_LISTENERS_PER_REQUEST = 50;
+
+const DEFAULT_PAYLOAD: ListenerPayload = {
+  flowNodeId: '',
+  pageSize: MAX_LISTENERS_PER_REQUEST,
+  searchAfter: [],
+  searchBefore: [],
+};
+
+class ProcessInstanceListeners {
+  state: State = {...DEFAULT_STATE};
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  startFetching = () => {
+    if (this.state.status === 'initial') {
+      this.state.status = 'first-fetch';
+    } else {
+      this.state.status = 'fetching';
+    }
+  };
+
+  startFetchingNext = () => {
+    this.state.status = 'fetching-next';
+  };
+
+  startFetchingPrev = () => {
+    this.state.status = 'fetching-prev';
+  };
+
+  handleFetchSuccess = () => {
+    this.state.status = 'fetched';
+  };
+
+  handleFetchFailure = () => {
+    this.state.status = 'error';
+    this.state.listenersCount = 0;
+    this.state.listeners = [];
+  };
+
+  setListeners = ({
+    listenersCount,
+    listeners,
+  }: {
+    listenersCount: number;
+    listeners: ListenerEntity[];
+  }) => {
+    this.state.listeners = listeners;
+    this.state.listenersCount = listenersCount;
+  };
+
+  setLatestFetchDetails = (fetchType: FetchType, listenersCount: number) => {
+    this.state.latestFetch.fetchType = fetchType;
+    this.state.latestFetch.itemsCount = listenersCount;
+  };
+
+  getListenersFailureCount = () => {
```

Comment:
This can be a [computed value](https://mobx.js.org/computeds.html).

```suggestion
  get listenersFailureCount() {
```

The benefit is that it would calculate the value only once (not on every method call).